pkg/agent/baker.go

```diff
@@ -450,9 +450,15 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			return cs.Properties.OrchestratorProfile.KubernetesConfig.RequiresDocker()
 		},
 		"HasDataDir": func() bool {
+			if profile != nil && profile.KubernetesConfig != nil && profile.KubernetesConfig.ContainerRuntimeConfig != nil && profile.KubernetesConfig.ContainerRuntimeConfig[datamodel.ContainerDataDirKey] != "" {
```

Comment:
@alexeldeib Do we have any data that the data dir key is set in cluster level only? We might have missed this before, but I feel it's a better and cleaner design to have these settings ONLY in agent pool level, it would be nice if we can move towards that direction. This is similar to the container runtime/teleport enabling discussion.

Reply:
Are you asking about the data for existing clusters with the toggle enabled or the behavior of the code? The current code does set it at cluster level, but I don't know how many users actually have that configured. 

My thinking was, if/when we expose this publicly, e.g. with https://dev.azure.com/msazure/CloudNativeCompute/_git/aks-rp/pullrequest/3659207 we could shift it to only be per-pool.

Reply:
+1 from me for making it pool only generally

Reply:
and yeah, this was done before Erik tackled containerd. I think the design for that is much cleaner (which I 100% copied with ephemeral os).